app/lib/widgets/cards/productCard/shared/price_discount.dart

```diff
@@ -0,0 +1,225 @@
+import 'package:flex_ui/widgets/cards/productCard/shared/price.dart';
+import 'package:flutter/material.dart';
+import 'package:widgetbook/widgetbook.dart';
+import 'package:widgetbook_annotation/widgetbook_annotation.dart' as widgetbook;
+import 'package:flex_ui/flex_ui.dart';
+import 'package:intl/intl.dart';
+
+/// This widget leverages the ``FlexPrice`` Widget for Price Formatting and display, but provides additional convenience for the conditional rendering of Sale Prices
+///
+/// FlexPriceDiscount implemented with same interface as FlexPrice with additional functionality
+/// - Conditional rendering of Current Price as a 'sale' price when an oldPrice is provided
+/// - When no 'oldPrice' is present, fallback to standard ``FlexPrice``
+/// - Theme customization for Price, Discount Price & Old Price values
+/// - Price & Old Price Semantic Label Optional Parameters
+/// - single line or line Wrap parameter for Price Value overflows
+///
+/// Parameters
+/// - [price]: Current Product Price, when [oldPrice] provided, will be styled as a 'sale price'
+/// - [oldPrice]: Previous Product Price. Default Text Styling linethrough
+/// - [priceFormatter]: Price Formatter Callback provided to underlying FlexPrice Widget
+/// - [priceStyle]: Override Price TextStyle for Current Price, Either when oldPrice is provided or not.
+/// - [oldPriceStyle]: Override [oldPrice] TextStyle (Default Strikethrough) FlexPrice widget
+/// - [priceLabel]: Text semantic label for [price]
+/// - [oldPriceLabel]: Text semantic label for [oldPrice]
+/// - [enableLineWrap]: Default false: single line display -TextOverflow.ellipsis, True: Enables Line wrap on overlfow
+///
+///
+/// Example usage:
+/// ```dart
+/// FlexPriceDiscount(
+///   price: 99.99,
+///   oldPrice: 200.00,
+///   priceStyle: TextStyle(color: Colors.blue), // Example optional style override
+///   priceFormatter: (price) => NumberFormat.currency(locale: 'en_US').format(price),
+///   priceLabel: "Sale Price {price}" // {price} token position indicates where the post formatter price should be placed in label
+/// )
+/// ```
+///
+class FlexPriceDiscount extends StatelessWidget {
+  const FlexPriceDiscount({
+    super.key,
+    required this.price,
+    this.oldPrice,
+    this.priceFormatter,
+    this.enableLineWrap = false,
```

Comment:
I'm back and forth on this boolean. Do you think there would be scenarios where we wouldn't want wrapping? Would the implementer just simply override more.. let's keep an eye on how we implement it in `flex_starter`? Open questions.

Reply:
I did as well, frankly would prefer to pull this out, default to wrap, and let implementer handle it -  would simplify the component a ton (No more need for the whole layout conditional Row/Wrap list building)

Was there for some edge case safeguarding like this likely unwanted wrap in our FlexProductCard for large product prices.
<img width="185" alt="image" src="https://github.com/user-attachments/assets/26ffc095-dc34-43fa-a402-4f8af421f1f9">


If we want to decide now it will make a few of the changes above unnecessary.

Reply:
https://github.com/flex-storefront/flex_ui/pull/19/commits/1366904a5d521c2b1da8db570e15c81094baa070 nixes enableLineWrap and all of the conditional layout building it required.